train.py

```diff
@@ -71,8 +72,9 @@ def discriminator_generator(it, g, dout_size=(16, 16)):
     # Sample fake and real pairs
     for a, b in it:
         a_fake = a
-        b_fake = g.predict(a_fake)
-
+        with graph1.as_default():

```

Comment:
There should be an check here for the backend. graph1 and graph2 are None otherwise.
Other than that I am ready to accept the PR.